runtime/shared_common/OSCache.cpp

```diff
@@ -107,8 +108,19 @@ SH_OSCache::getCacheVersionAndGen(J9PortLibrary* portlib, J9JavaVM* vm, char* bu
 	} else if (J9PORT_SHR_CACHE_TYPE_SNAPSHOT == versionData->cacheType) {
 		versionStr[versionStrLen] = J9SH_SNAPSHOT_PREFIX_CHAR;
 	}
+	if (generation < J9SH_GENERATION_37) {
+		j9str_printf(PORTLIB, genString, 4, "G%02d", generation);
+	} else if (J9SH_GENERATION_37 == generation) {
+		/* For generation 37, it is possible that there is no layer number in the cache file name */
+		if (layer < 0) {
+			j9str_printf(PORTLIB, genString, 4, "G%02d", generation);
+		} else {
+			j9str_printf(PORTLIB, genString, 7, "G%02dL%02d", generation, layer%((J9SH_LAYER_NUM_MAX_VALUE + 1)));
```

Comment:
Why the % mod operation? Perhaps an assert that the layer number is valid?

Reply:
Removed mod operation and added an assertion.